### PR TITLE
CSHARP-2431: Update start explicit encryption context.

### PR DESCRIPTION
**Pull request by DmitryLukyanov**

### CODE REVIEW

File: bindings/cs/MongoDB.Libmongocrypt.Test/BasicTests.cs [hunk 1]
```diff
@@ -186,7 +185,7 @@ public void EncryptBadBson()
         public void EncryptExplicit()
         {
             var keyDoc = ReadJsonTestFile("key-document.json");
-            Guid key = keyDoc["_id"].AsGuid;
+            var keyId = keyDoc["_id"].AsBsonBinaryData.Bytes;
```
Comment (DmitryLukyanov) [inline bindings/cs/MongoDB.Libmongocrypt.Test/BasicTests.cs:188]: NOTE: Steps to check it:
1. Update '_id' node in `key-document.json` :
     
```
"_id": {
    "$binary": {
      "base64": "LOCALAAAAAAAAAAAAAAAAA==",
      "subType": "04"
    }
}
```

2. Check the following:
```
keyDoc["_id"].AsBsonBinaryData.ToString()
>"UuidStandard:0x2ce0802c000000000000000000000000" // pay attention on the first bytes: 0x2ce0802
```
```
string.Join(", ", keyDoc["_id"].AsBsonBinaryData.Bytes)
"44, 224, 128, 44, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0"   // pay attention on the first bytes, they correspond with the the hex value in the previous comment
```
but the previous way(via `AsGuid`) provides wrong bytes order:
```
string.Join(", ", keyDoc["_id"].AsGuid.ToByteArray())
"44, 128, 224, 44, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0"  // pay attention on the first bytes, they has wrong order
```
even despite the fact that the string value is still the same as before:
```keyDoc["_id"].AsGuid.ToString()
"2ce0802c-0000-0000-0000-000000000000"
```

File: bindings/cs/MongoDB.Libmongocrypt.Test/BsonUtil.cs [hunk 2]
```diff
@@ -72,12 +71,12 @@ public static BsonDocument Concat(BsonDocument doc1, BsonDocument doc2)
 
         public static BsonDocument FromJSON(string str)
         {
-            using (var jsonReader = new JsonReader(str))
+            var jsonReaderSettings = new JsonReaderSettings { GuidRepresentation = GuidRepresentation.Unspecified };
```
Comment (DmitryLukyanov) [inline bindings/cs/MongoDB.Libmongocrypt.Test/BsonUtil.cs:74]: The same problem as it was in the driver.

**Review by vincentkam — APPROVED** [review]
LGTM! 😸  I'll push out a  new release onto myget

**Comment by DmitryLukyanov** [comment]
> 
> 
> LGTM! 😸 I'll push out a new release onto myget

Thanks!

**Comment by vincentkam** [comment]
Merged and pushed!

**Pull request closed without merging**